File: algorithms/102.swift
// https://leetcode.com/problems/binary-tree-level-order-traversal

/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public var val: Int
 *     public var left: TreeNode?
 *     public var right: TreeNode?
 *     public init(_ val: Int) {
 *         self.val = val
 *         self.left = nil
 *         self.right = nil
 *     }
 * }
 */

class Solution {
    func levelOrder(root: TreeNode?) -> [[Int]] {
        var result = [[Int]]()
        func dfs_(node: TreeNode?, _ level: Int) {
            guard let n = node else { return }
            if result.count < level+1 { result.append([]) }
            result[level].append(n.val)
            dfs_(n.left, level+1)
            dfs_(n.right, level+1)
        }
        dfs_(root, 0)
        return result
    }
}

File: algorithms/102.py
# https://leetcode.com/problems/binary-tree-level-order-traversal


# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution(object):
    def levelOrder(self, root):
        """
        :type root: TreeNode
        :rtype: List[List[int]]
        """

        result = []
        queue = []
        if root is not None:
            queue.append((0, root))

        while len(queue):
            level, node = queue.pop(0)
            if node.left is not None:
                queue.append((level + 1, node.left))
            if node.right is not None:
                queue.append((level + 1, node.right))
            if len(result) < level + 1:
                result.append([])
            result[level].append(node.val)
        return result
